### ISSUE TITLE
`json_encode` on empty Map returns array

### ISSUE THREAD

**Issue by petr-buchin** [php-ds/ext-ds#157]
I was glad to see, that `Map` data structure can be json-encoded. 
But if you run `json_encode()` on empty map, it gets gets encoded as an array - `[]`.
I think this is a bug, since Map should always be encoded as an object - `{}`.

Thank you very much in advance!

**Comment by rtheunissen** [comment]
This is unfortunately how [JsonSerializable](https://www.php.net/manual/en/jsonserializable.jsonserialize.php) works, where `jsonSerialize` returns the native value _to be encoded_ to JSON. 

The method in the extension is implemented to delegate it to an array:
```
METHOD(jsonSerialize)
{
    PARSE_NONE;
    ds_map_to_array(THIS_DS_MAP(), return_value);
}
```

Actually, we can cast that to an `(object)` which would encode as `{}`. It's unfortunate that this is O(n) to convert the map to an array first, then O(n) again to encode to JSON. I'd like to look into the possibility of a `toJson` method that interacts directly with a JSON api of some kind. 

I'll try the object cast in the meantime - thanks for brining this up.

**Comment by rtheunissen** [comment]
I think this should do it. https://github.com/php-ds/ext-ds/commit/078243f7519d39daeaad00edb422f5d0896703b2

**Comment by petr-buchin** [comment]
Thank you very much!

**Comment by rtheunissen** [comment]
This has now been fixed and released as part of 1.3.0
